### PR TITLE
Install adapters by cmake

### PR DESCRIPTION
**Pull request by lslusarczyk**
Add installing adapters. It is needed by https://github.com/oneapi-src/unified-runtime/pull/2146 and was already reviewed there and was just extracted to this separate PR (with some kind of test added).

### CODE REVIEW

**Comment by kbenzie** [comment]
With some additional changes in intel/llvm this could also resolve https://github.com/intel/llvm/issues/14924

File: .github/workflows/build-hw-reusable.yml [hunk 1]
```diff
@@ -87,13 +87,18 @@ jobs:
...
         -DUR_SYCL_LIBRARY_DIR=${{github.workspace}}/dpcpp_compiler/lib
+        -DCMAKE_INSTALL_PREFIX=${{github.workspace}}/install
         ${{ matrix.adapter.name == 'HIP' && '-DUR_CONFORMANCE_AMD_ARCH=gfx1030' || '' }}
         ${{ matrix.adapter.name == 'HIP' && '-DUR_HIP_PLATFORM=AMD' || '' }}
 
     - name: Build
       # This is so that device binaries can find the sycl runtime library
       run: cmake --build ${{github.workspace}}/build -j $(nproc)
 
+    - name: Install
```
Comment (lukaszstolarczuk) [inline .github/workflows/build-hw-reusable.yml:98]: looked into logs and it seems `gmock` and `gtest` is installed as well... I think it should rather not - if this is an easy fix you could add it here, if not probably just file an issue for that (the issue was here before your changes, of course)

**Review by lukaszstolarczuk — COMMENTED** [review]
👍

File: .github/workflows/build-hw-reusable.yml (hunk 1 above)
Comment (lslusarczyk) [reply .github/workflows/build-hw-reusable.yml:98]: Good point. Fixed.

File: .github/workflows/build-hw-reusable.yml [hunk 2]
```diff
@@ -87,13 +87,18 @@ jobs:
...
+      # This is to check that install command does not fail
+      run: cmake --install ${{github.workspace}}/build
+
```
Comment (pbalcer) [inline .github/workflows/build-hw-reusable.yml:101]: Ideally we'd check output against a list of expected files.

Comment (lslusarczyk) [reply .github/workflows/build-hw-reusable.yml:101]: I object. Too much logic to put in the test. Not worth doing it. Libs can be static or dynamic. Adapters may be built or not depending on cmake params. There are 100+ files installed, so check will have to selective and thus not complete.

If any client will depend on make install we can reconsider. Now the only client is benchmark framework.

Comment (pbalcer) [reply .github/workflows/build-hw-reusable.yml:101]: ok, makes sense.

**Comment by pbalcer** [comment]
If the changes are ready, please undraft them.

**Review by pbalcer — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]
Looks reasonable! 🚀

**Review by PietroGhg — APPROVED** [review]
Native CPU lgtm, thank you

**Review by kbenzie — APPROVED** [review]

**Pull request merged**